eng/pipelines/eng-common-sync.yml

```diff
@@ -30,16 +30,38 @@ stages:
         displayName: Sync ${{ parameters.DirectoryToSync }} Directory
 
         steps:
+        - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+          - pwsh: |
+              $patchDir = "$(Build.ArtifactStagingDirectory)/patchfiles"
```

Comment:
To help with diagnosing issues it might be worth publishing these patchfiles as an artifact so we can download and try them if something goes wrong.